main.js

```diff
@@ -383,13 +383,39 @@ function sortsch(sch){
 	sch.sort(function(a, b){if(b.d < a.d || (b.d == a.d && b.s < a.s)) return 1; return -1;});
 	return sch;
 }
+
+function calc_sleep_score(sch){
+	var sleep_score = 0;
+	for(var i=0; i < sch.length; i++){
+		if(sch[i].p != undefined){
+			sleep_score += sch[i].s;
+		}
+	}
+	return sleep_score;
+}
+
+function sort_for_sleep(){
+	for(var i=0; i<Math.min(10,schedules.length); i++){
+		var curMax = {"val":calc_sleep_score(schedules[i]), "index":i};
+
+		for(var j=i+1; j<schedules.length; j++){
+			if(calc_sleep_score(schedules[j]) > curMax.val){
+				curMax.val = calc_sleep_score(schedules[j]);
+				curMax.index = j;
+			}
+		}
+		[schedules[i],schedules[curMax.index]] = [schedules[curMax.index],schedules[i]];
```

Comment:
Eh, this swap is [considerably slower](https://jsperf.com/swap-array-vs-variable/42) than using a tmp.

Reply:
Okay, I'm fixing with using tmp.